### PR TITLE
Add helpful SiteService extension

### PR DESCRIPTION
**Pull request by MikeAlhayek**
Adding a helpful extension for `ISiteService`

```csharp
public static async Task<T> GetSettingsAsync<T>(this ISiteService siteService) where T : new()
{
    return (await siteService.GetSiteSettingsAsync()).As<T>();
}
```

### CODE REVIEW

File: src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Settings/SiteServiceExtensions.cs
Comment (hishamco) [inline src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Settings/SiteServiceExtensions.cs:10]:
```suggestion
    public static async Task<T> GetSettingsAsync<TSettings>(this ISiteService siteService) where TSettings : new()
        => (await siteService.GetSiteSettingsAsync()).As<TSettings>();
```

File: src/OrchardCore/OrchardCore.ReCaptcha.Core/Configuration/ReCaptchaSettingsConfiguration.cs
Comment (hishamco) [inline src/OrchardCore/OrchardCore.ReCaptcha.Core/Configuration/ReCaptchaSettingsConfiguration.cs:19]: How the format works this time :)

**Review by hishamco — CHANGES_REQUESTED** [review]

File: src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Settings/SiteServiceExtensions.cs
Comment (MikeAlhayek) [reply src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Settings/SiteServiceExtensions.cs:10]: Many people including @sebastienros and @Piedone do not like expression bodied methods. So I am defaulting to not use them any more.

Comment (MikeAlhayek) [reply src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Settings/SiteServiceExtensions.cs:10]: Not sure. This is just how VS formats the code and I keep it as is

Comment (hishamco) [reply src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Settings/SiteServiceExtensions.cs:10]: > Many people including @sebastienros and @Piedone do not like expression bodied methods. So I am defaulting to not use them any more.

It's not my or anyone else likes or dislikes, we should use a language feature whenever it's possible and fits

> Not sure. This is just how VS formats the code and I keep it as is

Yesterday you told me the VS doesn't add spaces when you move to second line, I need to check this myself

Comment (MikeAlhayek) [reply src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Settings/SiteServiceExtensions.cs:10]: Feel free to try the change in VS :) 

I would suggest opening a discussion about formatting not in this PR. The requested change does not add a value to the PR other than adhere to a personal preference. I don't mind method bodied methods. But, others seem to mind them. Opening a separate discussion would be a good start.

Comment (Piedone) [reply src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Settings/SiteServiceExtensions.cs:10]: Oh no, I actually like them, and at Lombiq we enforce them with code analysis too. It's just that in a prior exchange under a PR I pointed out, mistakenly, that we consciously don't use them in OC, but it turned out we actually do. Since then, I'm not disputing using them.

Also see to avoid such discussions in the future: https://github.com/OrchardCMS/OrchardCore/issues/7950#issuecomment-2138287937

Comment (MikeAlhayek) [reply src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Settings/SiteServiceExtensions.cs]: no problem here. I used expression bodied where possible.

File: src/OrchardCore/OrchardCore.ReCaptcha.Core/Configuration/ReCaptchaSettingsConfiguration.cs
Comment (MikeAlhayek) [reply src/OrchardCore/OrchardCore.ReCaptcha.Core/Configuration/ReCaptchaSettingsConfiguration.cs:19]: @hishamco did you try it out in VS?

Comment (hishamco) [reply src/OrchardCore/OrchardCore.ReCaptcha.Core/Configuration/ReCaptchaSettingsConfiguration.cs:19]: I tried the `main` branch moving a method call into another line adds 4 space, also using ternary condition operator too

Comment (MikeAlhayek) [reply src/OrchardCore/OrchardCore.ReCaptcha.Core/Configuration/ReCaptchaSettingsConfiguration.cs:19]: Does it place it on the same level as this PR does?

Comment (hishamco) [reply src/OrchardCore/OrchardCore.ReCaptcha.Core/Configuration/ReCaptchaSettingsConfiguration.cs:19]: I faced an issue with IIS Express once it fixed I will try it in this PR

File: src/OrchardCore.Modules/OrchardCore.Email/Migrations/EmailMigrations.cs
Comment (hyzx86) [inline src/OrchardCore.Modules/OrchardCore.Email/Migrations/EmailMigrations.cs:30]:
```suggestion
            var siteService = scope.ServiceProvider.GetRequiredService<ISiteService>();
            var smtpSettings = await siteService.GetSettingsAsync<SmtpSettings>();
```

File: src/OrchardCore/OrchardCore.ReCaptcha.Core/Configuration/ReCaptchaSettingsConfiguration.cs
Comment (MikeAlhayek) [reply src/OrchardCore/OrchardCore.ReCaptcha.Core/Configuration/ReCaptchaSettingsConfiguration.cs:19]: @hishamco I do not want to want too much on this PR. Every merge will cause a conflict :) So please either approve or merge this PR once you finish what every you need to check.

Comment (MikeAlhayek) [reply src/OrchardCore/OrchardCore.ReCaptcha.Core/Configuration/ReCaptchaSettingsConfiguration.cs:19]: another conflict today @hishamco? Are you really checking this spacing issue? I am asking this because it seems to be a time wasting step.

Comment (hishamco) [reply src/OrchardCore/OrchardCore.ReCaptcha.Core/Configuration/ReCaptchaSettingsConfiguration.cs:19]: Mike, I'm not sure why you are in a rush in almost all the PRs :) as you know the time zones are different and some of us working on other PRs, issues or something else

Believe me it happen in almost OSS, I will start checking you PR today

File: src/OrchardCore.Modules/OrchardCore.Email.Smtp/Services/SmtpOptionsConfiguration.cs
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Email.Smtp/Services/SmtpOptionsConfiguration.cs:30]: As I said in another PR the VS adds 4 spaces when you move the method call in the next line

/cc @hyzx86

File: src/docs/releases/2.0.0.md
Comment (hishamco) [inline src/docs/releases/2.0.0.md:521]:
```suggestion
A new extension method named `GetSettingsAsync<T>()` has been added to the `ISiteService` interface. This method allows you to retrieve specific settings with a single line of code. For example, to get the `LoginSettings`, you can now use:
```

**Review by hishamco — APPROVED** [review]
Please react to the suggestion and comment before merge

File: src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Settings/SiteServiceExtensions.cs
Comment (MikeAlhayek) [reply src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Settings/SiteServiceExtensions.cs:8]: It could return null as it does today. I don't want to change the behavior

File: src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Settings/SiteServiceExtensions.cs
Comment (MikeAlhayek) [reply src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Settings/SiteServiceExtensions.cs:5]: What do you like to add? Please add using suggestion. The method is self explanatory

File: src/OrchardCore/OrchardCore.ReCaptcha.Core/Configuration/ReCaptchaSettingsConfiguration.cs
Comment (MikeAlhayek) [reply src/OrchardCore/OrchardCore.ReCaptcha.Core/Configuration/ReCaptchaSettingsConfiguration.cs:19]: Because of too many conflicts. + I don't see a value is blocking a PR because you do not agree with VS formatting :) Let's focus on valuable feedback so we are not spending too much time debating or resolving conflicts that can be simply avoided. I hope you understand my point.

File: src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Settings/SiteServiceExtensions.cs
Comment (hishamco) [reply src/OrchardCore/OrchardCore.Infrastructure.Abstractions/Settings/SiteServiceExtensions.cs:14]: I didn't ask to change the behavior, I just need to be sure

**Pull request merged**
